### PR TITLE
add element deprecation documentation

### PR DESCRIPTION
**Pull request by brynnz22**
Regarding: https://github.com/microbiomedata/nmdc-schema/issues/1314

I added @turbomam words almost exactly from the issue with some minor adjustments.
- I included @cmungall suggestion of including a reason for the deprecation and a link to the corresponding issue. I suggested doing this using the `deprecated` slot (since it takes a `string`) in addition to the replacement slots (they take `uriorcurie`). Let me know if you think this should be done another way.
- I added commenting out all element usages prior to fully removing them, so we can still understand where they are being used for one release cycle in case it becomes important.

### CODE REVIEW

File: src/docs/schema_element_deprecation_guide.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Deprecating NMDC schema elements
+
+When deprecating schema elements, it is a two step process:
+
+1. For one release cycle, the schema will include deprecation annotations to the element's definition and commenting out any usages of the slot (e.g. in a class, domain, range, etc) including a deprecation date (e.g. `# doi_awards deprecated on 11/12/2023` in a class's usage)
```
Comment (turbomam) [inline src/docs/schema_element_deprecation_guide.md:5]: let's encourage our teammates to use iso8601/international dates like 2023-11-12

**Review by turbomam — CHANGES_REQUESTED** [review]
thanks, this is very valuable

exposing new content like this also requires that the filename is added to the `nav` section of `mkdocs.yml`

File: src/docs/schema_element_deprecation_guide.md (hunk 1 above)
Comment (brynnz22) [reply src/docs/schema_element_deprecation_guide.md]: I added this.

**Comment by turbomam** [comment]
I am going to check this out, add `src/docs/schema_element_deprecation_guide.md`  to the `nav` section of `mkdocs.yml`, and then merge.

**Review by turbomam — APPROVED** [review]

**Pull request merged**